core/src/main/java/org/apache/spark/io/LZ4BlockInputStream.java

```diff
@@ -197,7 +197,7 @@ private void refill() throws IOException {
         readFully(buffer, originalLen);
         break;
       case COMPRESSION_METHOD_LZ4:
-        if (compressedBuffer.length < originalLen) {
+        if (compressedBuffer.length < compressedLen) {
```

Comment:
Does this possibly happen?

I go to check https://github.com/jpountz/lz4-java/blob/master/src/java/net/jpountz/lz4/LZ4BlockOutputStream.java#L192

If the compressed lengh is more the original length, it will choose `COMPRESSION_METHOD_RAW` as compress method, instead of `COMPRESSION_METHOD_LZ4`. In other words, the compressed length is never more then original length under `COMPRESSION_METHOD_LZ4`.